public/app/core/components/NestedFolderPicker/NestedFolderPicker.test.tsx

```diff
@@ -174,14 +167,36 @@ describe('NestedFolderPicker', () => {
   });
 
   it('hides folders specififed by UID', async () => {
-    render(<NestedFolderPicker excludeUIDs={[folderA.item.uid]} onChange={mockOnChange} />);
+    render(<NestedFolderPicker excludeUIDs={[folderC.item.uid]} onChange={mockOnChange} />);
 
     // Open the picker and wait for children to load
     const button = await screen.findByRole('button', { name: 'Select folder' });
     await userEvent.click(button);
-    await screen.findByLabelText(folderB.item.title);
+    await screen.findByLabelText(folderA.item.title);
 
-    expect(screen.queryByLabelText(folderA.item.title)).not.toBeInTheDocument();
+    expect(screen.queryByLabelText(folderC.item.title)).not.toBeInTheDocument();
+  });
+
+  it('by default only shows items the user can edit', async () => {
+    render(<NestedFolderPicker onChange={mockOnChange} />);
+
+    const button = await screen.findByRole('button', { name: 'Select folder' });
+    await userEvent.click(button);
+    await screen.findByLabelText(folderA.item.title);
```

Comment:
really minor, what do you think about wrapping this in an `expect` same as the others? e.g.
```
expect(await screen.findByLabelTest(folderA.item.title)).toBeInTheDocument();
```

Reply:
I would rather not - this isn't really supposed to be an assertion (doing this in all the tests isn't great), but rather is a proxy for "wait until the picker has open and loaded".

Reply:
yeah i get that, but you presumably also want an assertion that `folderA` is present and visible (as you have for `folderC` for example)? 🤔 happy for you to merge either way 🙌